docs/gen_ref_pages.py

```diff
@@ -1,111 +1,46 @@
 """
 Generate the code reference pages.
 
-Source: https://mkdocstrings.github.io/recipes/
+This script traverses the markdown files under docs/src/reference and for each:
+- If the file is not empty it is ignored and will appear in the docs as-is
+    (after mkdocstrings has filled in any identifiers inside it).
+- If the file is empty, it is interpreted as a placeholder for the top-level
+    symbol with its same name and its reference page is generated in its same
+    path. E.g., if an empty docs/src/reference/temporian/io/to_csv.md file
+    exists, the reference page for `temporian.to_csv` will be generated under
+    reference/temporian/io/to_csv/ in the docs.
+
+Related: https://mkdocstrings.github.io/recipes/
 """
 
 from pathlib import Path
-from typing import Set, Tuple
 
 import mkdocs_gen_files
 
 nav = mkdocs_gen_files.Nav()
 
-SRC_PATH = Path("temporian")
+REFERENCE = Path("docs/src/reference")
 
-# Stores symbol and path of each public API member
-members: Set[Tuple[str, Path]] = set()
+# Sort files by depth and name for alphabetical order and subdirs at the bottom
+paths = sorted(
+    REFERENCE.rglob("*.md"), key=lambda p: str(len(p.parts)) + str(p)
+)
 
-non_parsable_imports = []
+for path in paths:
+    path = Path(path)
+    ref_path = path.relative_to(REFERENCE)
+    nav_path = ref_path.with_suffix("")
+    full_ref_path = "reference" / ref_path
 
-# We need to be able to parse other files to allow wildcard imports
-# Storing pair of (prefix, path) to parse in a stack
-files_to_parse = [(None, SRC_PATH / "__init__.py")]
+    nav[list(nav_path.parts)] = ref_path.as_posix()
 
-while files_to_parse:
-    prefix, file = files_to_parse.pop()
+    # If file is empty we assume it's a top-level symbol and auto
+    # generate the mkdocstring identifier for it
+    if path.stat().st_size == 0:
+        with mkdocs_gen_files.open(full_ref_path, "w") as fd:
+            ident = "temporian." + nav_path.name
+            fd.write(f"::: {ident}")
```

Comment:
Are we using the file if it's not empty?
Can we have a print to show the list of non-empty files that have been found?

I'm mostly worried about potentially silent bugs generating some empty page.

Reply:
added print of non-empty files found